libs/gi/ui/src/components/character/CharacterMultiSelectionModal.tsx

```diff
@@ -0,0 +1,500 @@
+import CloseIcon from '@mui/icons-material/Close'
+import FavoriteIcon from '@mui/icons-material/Favorite'
+import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder'
+import type { TooltipProps } from '@mui/material'
+import {
+  Box,
+  CardActionArea,
+  CardContent,
+  Divider,
+  Grid,
+  IconButton,
+  TextField,
+  Tooltip,
+  Typography,
+  styled,
+  tooltipClasses,
+} from '@mui/material'
+import type { ChangeEvent } from 'react'
+import {
+  useContext,
+  useDeferredValue,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
+import { useTranslation } from 'react-i18next'
+import { DataContext, SillyContext } from '../../context'
+import { iconAsset } from '../../util/iconAsset'
+import { ElementToggle, WeaponToggle } from '../toggles'
+import { CharacterCard } from './CharacterCard'
+import type { CharacterSortKey } from './CharacterSort'
+import {
+  characterFilterConfigs,
+  characterSortConfigs,
+  characterSortMap,
+} from './CharacterSort'
+import { CharacterName } from './Trans'
+
+type characterFilter = (
+  characterKey: CharacterKey,
+  character: ICachedCharacter | undefined,
+  sheet: CharacterSheet
+) => boolean
+
+type CharacterMultiSelectionModalProps = {
+  show: boolean
+  newFirst?: boolean
+  onHide: () => void
+  onMultiSelect?: (cKeys: (CharacterKey | '')[]) => void
+  teamId: string
+  filter?: characterFilter
+}
+const sortKeys = Object.keys(characterSortMap)
+export function CharacterMultiSelectionModal({
+  show,
+  onHide,
+  onMultiSelect,
+  teamId,
+  filter = () => true,
+  newFirst = false,
+}: CharacterMultiSelectionModalProps) {
+  const { t } = useTranslation([
+    'page_character',
+    // Always load these 2 so character names are loaded for searching/sorting
+    'sillyWisher_charNames',
+    'charNames_gen',
+  ])
+  const { silly } = useContext(SillyContext)
+  const database = useDatabase()
+  const [state, setState] = useState(() => database.displayCharacter.get())
+  useEffect(
+    () => database.displayCharacter.follow((r, s) => setState(s)),
+    [database, setState]
+  )
+
+  const team = database.teams.get(teamId)!
+  const { loadoutData } = team
+  const currentTeamCharKeys : (CharacterKey | '')[] = useMemo(() => {
+    const keys = [] as (CharacterKey | '')[]
+    loadoutData.forEach((loadoutDatum) => {
+      keys.push(database.teamChars.get(loadoutDatum?.teamCharId)?.key ?? '')
+    })
+    return keys
+  }, [database, loadoutData])
+
+  const { gender } = useDBMeta()
+
+  const [dbDirty, forceUpdate] = useForceUpdate()
+
+  // character favorite updater
+  useEffect(
+    () => database.charMeta.followAny(() => forceUpdate()),
+    [forceUpdate, database]
+  )
+
+  const [searchTerm, setSearchTerm] = useState('')
+  const deferredSearchTerm = useDeferredValue(searchTerm)
+  const deferredState = useDeferredValue(state)
+  const deferredDbDirty = useDeferredValue(dbDirty)
+  const characterKeyList = useMemo(() => {
```

Comment:
I think you need to keep the selected characters to be displayed at all times, even if there is a filter/search